### PR TITLE
Fix video_core includes (and dependencies) using include-what-you-use

### PR DESCRIPTION
**Pull request by linkmauve**
This has only be tested on gcc 5.3 and clang 3.7 currently.

### CODE REVIEW

**Comment by Subv** [comment]
It seems to pass the MSVC build test successfully

**Comment by linkmauve** [comment]
I just did a test run with both compilers on my i5-3210m, the changes aren’t high but are statistically significant (2.57% and 2.43% improvement respectively for gcc and clang):

```
gcc 5.3.0:
make -j4  502.56s user 27.26s system 382% cpu 2:18.60 total

old gcc 5.3.0:
make -j4  513.56s user 27.83s system 380% cpu 2:22.25 total

clang 3.7.1:
make -j4  419.78s user 11.81s system 382% cpu 1:52.90 total

old clang 3.7.1:
make -j4  431.44s user 12.05s system 383% cpu 1:55.71 total
```

**Comment by Subv** [comment]
Visual Studio can generate them, but I don't know if there's a way to do that with Clang or GCC.

**Comment by MerryMage** [comment]
Include graphs, before and after:

https://gist.github.com/MerryMage/c77948b18544ac875586be32662dc826

10712 includes → 9124 includes (15% improvement)

File: src/video_core/pica.h [hunk 1]
```diff
@@ -5,19 +5,20 @@
 #pragma once
 
 #include <array>
-#include <cmath>
 #include <cstddef>
 #include <string>
 
+#ifndef _MSC_VER
```
Comment (MerryMage) [inline src/video_core/pica.h]: I don't think one should exclude MSVC here.

Comment (linkmauve) [inline src/video_core/pica.h]: This header is only used on line 41 (and only if the includer used the PICA_REG_INDEX_WORKAROUND macro), so only on ¬MSVC.

Comment (MerryMage) [inline src/video_core/pica.h]: Ah, makes sense, apologies.

**Comment by MerryMage** [comment]
LGTM

**Comment by bunnei** [comment]
LGTM

**Pull request merged**